### PR TITLE
feat: configuration pallet

### PR DESCRIPTION
**Pull request by weichweich**
## related https://github.com/KILTprotocol/ticket/issues/2522

This PR adds a configuration pallet that dynamically enables strict blocknumber checks.

## Checklist:

- [x] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: pallets/pallet-configuration/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+	use cumulus_pallet_parachain_system::{CheckAssociatedRelayNumber, RelayNumberStrictlyIncreases};
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{EnsureOriginWithArg, StorageVersion},
+	};
+	use frame_system::pallet_prelude::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```
Comment (ntn-x2) [inline pallets/pallet-configuration/src/lib.rs:57]: This should be 0, which is also the default and does not need to be written to storage for new pallets.

File: pallets/pallet-configuration/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			<T as pallet::Config>::EnsureOrigin::ensure_origin(origin, &configuration)?;
+
+			ConfigurationStore::<T>::set(configuration.clone());
+			Self::deposit_event(Event::<T>::ConfigurationUpdate(configuration));
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> CheckAssociatedRelayNumber for Pallet<T> {
```
Comment (ntn-x2) [inline pallets/pallet-configuration/src/lib.rs:104]: I think the pallet still has to be used in the cumulus runtime config

Comment (weichweich) [reply pallets/pallet-configuration/src/lib.rs:104]: [27f3605](https://github.com/KILTprotocol/kilt-node/pull/522/commits/27f3605c493f33654099475867b915a388080ed2)

File: pallets/pallet-configuration/src/lib.rs (hunk 1 above)
Comment (weichweich) [reply pallets/pallet-configuration/src/lib.rs]: [27f3605](https://github.com/KILTprotocol/kilt-node/pull/522/commits/27f3605c493f33654099475867b915a388080ed2)

File: runtimes/peregrine/src/lib.rs [hunk 3]
```diff
@@ -527,6 +527,12 @@ impl pallet_tips::Config for Runtime {
 	type WeightInfo = weights::pallet_tips::WeightInfo<Runtime>;
 }
 
+impl pallet_configuration::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
```
[the remark this answers is not in the document]
Comment (weichweich) [reply runtimes/peregrine/src/lib.rs]: [27f3605](https://github.com/KILTprotocol/kilt-node/pull/522/commits/27f3605c493f33654099475867b915a388080ed2)

**Review by ntn-x2 — APPROVED** [review]

**Pull request merged**